One Click Install Alpha

Geenz opened the pull request:
as of = 2026-01-26 for PV deploy
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project%2385298ebb-OneClickInstall
cohort = One Click Install
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project%2385298ebb-OneClickInstall
desired = 0
relnotes:

# One Click Install Alpha Viewer

Hey everyone!

This release we have the One Click Install Alpha Viewer.  This is an early alpha release to streamline the experience of setting up the viewer on Windows computers and getting in world.  This also will help ease the update process for viewers moving forward.

## What's different

* Viewers on Windows now use a new "one click" installer that installs the viewer to a local user directory.
* Viewers on macOS still use the disk image method for drag and drop install.
* All viewers will now use a new updater that makes updates more transparent.
  * Viewers will no longer prompt the user to install an update.  It will now install in the background.
* Installing to custom locations is supported by the `--installto` command line parameter
* Portable installations can be found on our GitHub: https://github.com/secondlife/viewer/releases/tag/Second_Life_Project%2385298ebb-OneClickInstall

At present, you will have two viewers installed: the one click viewer and the current viewer.  This is expected, and will be improved in a future alpha.

Geenz committed:
Integrate Velopack installer and update framework

Note: Updates don't quite work yet.

Geenz committed:
Try to get CI/CD working with velopack.

Geenz committed:
Update build.yaml

Geenz committed:
Update build.yaml

Geenz committed:
Update build.yaml

kurtlinden commented:
Tested on build Second Life Project One-click 26.1.0.20387239236 

Test plan provided to QA:

* Does it install to [user]/AppData/Local
* Does it run
* Does it have an uninstaller in add/remove programs

I verified on Windows machine that the build installs with one click to [user]/AppData/Local and after install the viewer launches and runs successfully.  Viewer is installed Second Life Project One-Click

I verified that the app can be removed in the add/remove apps section and the viewer is deleted and leaves behind a SecondLifeProjectOne-Click folder in [user]/AppData/Local containing just 2 files .dead and velopack (log?)

I verified you could successfully relaunch the app from the desktop icon and from the App list on the start menu.

I verified the Mac build installs on Mac the same way current release viewer does successfully

Issues noticed in testing.

* After the viewer launches first time after install a updater error dialog shows https://gyazo.com/4d9e5a2c4b20e6179d449f1ec97afd92 which can be dismissed and login continues normally
* On second login to the viewer, the desktop icon for the Second Life Project One-Click gets removed from the desktop and the same updater error shows.  This is caused by the updater's delete_wrapper firing. It's meant to clean old local installs. velopack is a local install.
* On third login to the viewer no more update dialog errors show

So in the spirit of wanting to get the one click install to get user feedback this seems to be meet those goals but the issues should be identified in the release notes to help out the users trying it out.

Passed QA based on test plan provided and goals of this release.

kurtlinden commented:
Per discussion with grumpity, to make a good first impression with this the updater issues should be addressed before release.

Hecklezz commented on indra/newview/llvelopack.cpp:
```diff
@@ -0,0 +1,379 @@
+
+static const wchar_t* PROTOCOL_SECONDLIFE = L"secondlife";
+static const wchar_t* PROTOCOL_GRID_INFO = L"x-grid-location-info";
+static const wchar_t* VIEWER_EXE_NAME = L"SecondLifeViewer.exe";
```
This to me seems unnecessary to hardcode the viewer exe name when it can be obtained from gDirUtilp->getExecutableFilename() and then use ll_convert\<std::wstring\>, since LLDir_Win32 will have been constructed already before WINMAIN has been called.

This would allow for better compatibility with TPV's so they don't all need to change this hardcoded constant to work with Velopack.

For sanity sake, I tested with my development build of Firestorm and after it ran the Velopack installer, inside on_after_install using gDirUtilp->getExecutableFilename() returned the correct exe name, the viewer worked correctly and the installer setup the shortcuts and protocols correctly too.

Geenz committed:
Add Velopack update support for macOS and VVM integration

Geenz committed:
Update Velopack version and dependencies

Geenz committed:
Update viewer_manifest.py

Geenz committed:
Improve Velopack packaging for macOS

Geenz committed:
Update build.yaml

Geenz committed:
Update viewer_manifest.py

Geenz committed:
Update viewer_manifest.py

Geenz committed:
Update viewer_manifest.py

Geenz committed:
Update viewer_manifest.py

Geenz committed:
Update viewer_manifest.py

Dan-Linden commented:
|  One Click Install Alpha 26.1.0.21295806042 | Result | Comment |
|---|---|---|
| secondlife/iqa#596 Install Test | Passed | Passed |
| **Overall Result** | **Passed** | **OK to ship Alpha** |

**NOTES**
Only installing and uninstalling was tested.
Updater tests did **not** pass. Updating this viewer may not be possible.

Anti-virus false positive report:
| File | Security vendor analysis |
|---|---|
| SLVersionChecker.exe | SecureAge - Malicious <br>Skyhigh (SWG) - BehavesLike.Win64.Generictrojan.tc  |
| slplugin.exe | MaxSecure - Trojan.Malware.300983.susgen<br>SecureAge - Malicious |

kylelinden reviewed (approved):
One Click Install Alpha approved
viewer = 26.1.0.21295806042
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project%2385298ebb-OneClickInstall
cohort = One Click Install
desired = 0

Geenz commented:
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project%2385298ebb-OneClickInstall

Hecklezz commented on indra/newview/llvelopack.cpp:
```diff
@@ -0,0 +1,437 @@
+    register_protocol_handler(PROTOCOL_SECONDLIFE, L"URL:Second Life", exe_path);
+    register_protocol_handler(PROTOCOL_GRID_INFO, L"URL:Second Life", exe_path);
+    register_uninstall_info(install_dir, app_name, version);
+    create_shortcuts(install_dir, app_name);
```
Velopack already creates both the Desktop and Start Menu shortcuts based on the app name given to the --packTitle argument in the vpk call, it also uninstalls them for you as well. 
So calling create_shortcuts here and the remove_shortcuts a little further below to do them manually, seems redundant?

The only difference I noticed, is that Velopack by default doesn't put the start menu shortcut inside of it's own folder, but the desktop shortcut is the same. Which also means, currently the installer creates two different start menu shortcuts (one in Start Menu\Programs\Second Life Test, the other just in Start Menu\Programs)
The velopack start menu shortcut location can be changed with the --shortcuts argument in the vpk call.

Geenz replied:
Resolved in 26.1.1 - we just don't generate shortcuts with VPK now.  We generate with the post install handler in the viewer.

akleshchev committed:
#5346 Uninstall older non-velopack viewer (#5363)

* #5335 Fix silent uninstall asking about registry

* #5346 Uninstall older non-velopack viewer

if of the same channel

Geenz committed:
Merge branch 'release/2026.01' into project/one-click

Geenz committed:
Use runtime viewer exe name, handle Velopack URL

akleshchev committed:
Velopack download failure diagnostic (#5520)

* Velopack download failure diagnostic

* Fix up velopack downloading updates.  Handle updates internally then hand them off to velopack. (#5524)

* Update llvelopack.cpp

* More velopack changes.  Should download updates properly now.

* Update llvvmquery.cpp

* Don't include NSI files

* Restore optional updates, refine viewer restart behavior. (#5527)

* Add support for optional updates.

* Don't restart the viewer after the update unless it was optional.

* Setup UpdaterServiceSetting with velopack properly.

* Refine the restart behavior a bit - readd the old "the viewer must update" UX.

* If the update is still downloading, close should just reopen the downloading dialog.

Also add a login guard - probably unnecessary, but you never know with how creative our residents get.

---------

Co-authored-by: Jonathan "Geenz" Goodman <geenz@lindenlab.com>

Geenz committed:
Remove SLVersionChecker from the viewer with velopack. (#5528)

* Remove SLVersionChecker updater integration

* Ensure that the portable install has the correct version number.

Geenz committed:
Merge pull request #5529 from secondlife/release/2026.01

Merge  2026.1 into one click

Geenz commented:
Closing in favor of #5530
